backend/infrahub/graphql/auth/query_permission_checker/object_permission_checker.py

```diff
@@ -71,3 +75,47 @@ async def check(
                 raise PermissionDeniedError(f"You do not have the following permission: {permission}")
 
         return CheckerResolution.NEXT_CHECKER
+
+
+class AccountManagerPermissionChecker(GraphQLQueryPermissionCheckerInterface):
+    """Checker that makes sure a user account perform actions on account related objects.
+
+    This is similar to object permission checker except that we care for any operations on any account related kinds.
+    """
+
+    permission_required = f"global:{GlobalPermissions.MANAGE_ACCOUNTS.value}:{PermissionDecision.ALLOW.value}"
+
+    async def supports(self, db: InfrahubDatabase, account_session: AccountSession, branch: Branch) -> bool:
+        return account_session.authenticated
+
+    async def check(
+        self,
+        db: InfrahubDatabase,
+        account_session: AccountSession,
+        analyzed_query: InfrahubGraphQLQueryAnalyzer,
+        query_parameters: GraphqlParams,
+        branch: Branch,
+    ) -> CheckerResolution:
+        has_permission = False
+        for permission_backend in registry.permission_backends:
+            if has_permission := await permission_backend.has_permission(
+                db=db, account_id=account_session.account_id, permission=self.permission_required, branch=branch
+            ):
+                break
+
+        is_account_operation = False
+        kinds = await analyzed_query.get_models_in_use(types=query_parameters.context.types)
+
+        for kind in kinds:
+            schema = get_schema(db=db, branch=branch, node_schema=kind)
+            if is_account_operation := kind in (
+                InfrahubKind.GENERICACCOUNT,
+                InfrahubKind.ACCOUNTGROUP,
+                InfrahubKind.ACCOUNTROLE,
+            ) or (isinstance(schema, NodeSchema) and InfrahubKind.GENERICACCOUNT in schema.inherit_from):
+                break
```

Comment:
could you do the `is_account_operation` logic before the `has_permission` logic and then return early if `is_account_operation` is False?